### PR TITLE
fix share ptr convert error

### PR DESCRIPTION
**Pull request by findmyway**
fix https://github.com/deepmind/open_spiel/issues/159

### CODE REVIEW

**Comment by findmyway** [comment]
Hi @lanctot ,

This is just a quick fix. I report the error here https://github.com/JuliaInterop/CxxWrap.jl/issues/212 

By the way, do you have any timeline to tag a release? I have registered `OpenSpiel` and `OpenSpiel_jll` based on a commit several weeks ago in Julia, so that users can simply run `] add OpenSpiel` in Julia REPL to use this package (but it only works in Linux for now). Could you kindly ping me a week ahead so that I can update the doc in time (and hopefully make it also work on Mac 
 at that time)

**Comment by lanctot** [comment]
Wow, amazing response time, thanks! Though that fix is really not obvious! It's good to have the example in case we run into it again.

Tagging a release is a good idea. We have talked about it a lot but not done it. Let me discuss it with the team but I expect we wil do it in the coming months. I will give you a heads up when we do.

**Comment by findmyway** [comment]
> It's good to have the example in case we run into it again.

Yeah, [the following line](https://github.com/deepmind/open_spiel/pull/160/files#diff-249d595fcdaf07842d8bfbd6afa0f40fR1) is the only relevant change:

```julia
Base.convert(::Type{CxxWrap.StdLib.SharedPtrAllocated{Evaluator}}, ::CxxWrap.StdLib.SharedPtrAllocated{Evaluator}) = p
```

Just replace `Evaluator` to some other type the next time. There should be a more elegant approach. But I just can't figure it out right now. Let's see if the author of CxxWrap.jl has any thoughts.

>  I will give you a heads up when we do.

Thanks!

**Comment by lanctot** [comment]
I have imported this PR, should be merged on Monday's update.

File: open_spiel/examples/gtp.cc [hunk 1]
```diff
@@ -40,7 +40,7 @@ std::unique_ptr<open_spiel::Bot> MakeBot(
     const open_spiel::Game& game,
     std::shared_ptr<open_spiel::algorithms::Evaluator> evaluator) {
   return std::make_unique<open_spiel::algorithms::MCTSBot>(
-      game, std::move(evaluator), /*uct_c=*/2, /*max_simulations=*/1000,
```
Comment (tkoeppe) [inline open_spiel/examples/gtp.cc]: This is a strict pessimisation, could you kindly explain a bit why this is necessary?

Comment (findmyway) [reply open_spiel/examples/gtp.cc]: Hi @tkoeppe ,

I think that I brought in this change by accident. During development, there was a build error without this change. But I just checked it again and found that it was unnecessary here.

Thanks for pointing this out! Do I need to make another PR to revert this change, or could you just modify it directly?

Comment (tkoeppe) [reply open_spiel/examples/gtp.cc]: Let's leave that to @lanctot to decide, thanks!

**Comment by lanctot** [comment]
Let's do it from our side. Thanks @tkoeppe, I learned something about `shared_ptr` today!

**Pull request merged**